File: frontend/src/lib/api/alerts.ts
import { apiFetch } from "./client";

export type Alert = {
  id: string;
  user_id: string;
  project_id: string | null;
  alert_type: string;
  severity: "critical" | "warning" | "info" | "success";
  title: string;
  message: string;
  action_url: string | null;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
};

export async function listAlerts(params?: { is_read?: boolean; severity?: string }): Promise<Alert[]> {
  const p: Record<string, string> = {};
  if (params?.is_read !== undefined) p.is_read = String(params.is_read);
  if (params?.severity) p.severity = params.severity;
  return apiFetch<Alert[]>("/api/alerts", { params: p });
}

export async function markAlertRead(id: string): Promise<void> {
  return apiFetch(`/api/alerts/${id}/read`, { method: "PATCH" });
}

export async function markAllAlertsRead(): Promise<void> {
  return apiFetch("/api/alerts/read-all", { method: "POST" });
}

File: backend/app/api/routes/alerts.py
"""Alert management."""

from fastapi import APIRouter, Depends

from app.api.dependencies.auth import get_current_user
from app.core.supabase import supabase

router = APIRouter(prefix="/alerts", tags=["alerts"])


@router.get("")
async def list_alerts(
    is_read: bool | None = None,
    severity: str | None = None,
    user: dict = Depends(get_current_user),
):
    query = (
        supabase.table("alerts")
        .select("*")
        .eq("user_id", user["id"])
        .order("created_at", desc=True)
        .limit(50)
    )
    if is_read is not None:
        query = query.eq("is_read", is_read)
    if severity:
        query = query.eq("severity", severity)
    result = query.execute()
    return result.data


@router.patch("/{alert_id}/read")
async def mark_read(alert_id: str, user: dict = Depends(get_current_user)):
    supabase.table("alerts").update({
        "is_read": True,
        "read_at": "now()",
    }).eq("id", alert_id).eq("user_id", user["id"]).execute()
    return {"ok": True}


@router.post("/read-all")
async def mark_all_read(user: dict = Depends(get_current_user)):
    supabase.table("alerts").update({
        "is_read": True,
        "read_at": "now()",
    }).eq("user_id", user["id"]).eq("is_read", False).execute()
    return {"ok": True}
